sycl/source/handler.cpp

```diff
@@ -111,9 +111,10 @@ bool handler::isStateExplicitKernelBundle() const {
 std::shared_ptr<detail::kernel_bundle_impl>
 handler::getOrInsertHandlerKernelBundle(bool Insert) const {
   if (!MImpl->MKernelBundle && Insert) {
-    MImpl->MKernelBundle =
-        detail::getSyclObjImpl(get_kernel_bundle<bundle_state::input>(
-            MQueue->get_context(), {MQueue->get_device()}, {}));
+    auto Ctx = MGraph ? MGraph->getContext() : MQueue->get_context();
+    auto Dev = MGraph ? MGraph->getDevice() : MQueue->get_device();
```

Comment:
```suggestion
    auto &Ctx = MGraph ? MGraph->getContext() : MQueue->get_context();
    auto &Dev = MGraph ? MGraph->getDevice() : MQueue->get_device();
```
If both return references? Same applies other places if applicable here.

Reply:
None of these actually return references, the first returns a `sycl::context` and the second `sycl::device`, both by value.